modules/nf-core/glimpse2/phase/main.nf

```diff
@@ -0,0 +1,69 @@
+process GLIMPSE2_PHASE {
+    tag "$meta.id"
+    label 'process_medium'
+
+    beforeScript  """
+    if cat /proc/cpuinfo | grep avx2 -q
+    then
+        echo "Feature AVX2 present"
+    else
+        echo "Feature AVX2 not present on node"
+        exit 1
+    fi
+    """
+
+    conda "bioconda::glimpse-bio=2.0.0"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/glimpse-bio:2.0.0--hf340a29_0':
+        'quay.io/biocontainers/glimpse-bio:2.0.0--hf340a29_0' }"
+
+    input:
+        tuple val(meta), path(input), path(input_index), val(input_region), val(output_region), path(reference), path(reference_index), path(map), path(samples_file)
```

Comment:
Can you exclude all optional value parameters from here? (like input_region and value_region)

Reply:
So it's mandatory but only in some cases if I get that correctly?

Reply:
Sorry I wasn't clear.
They are always mandatory.
The software is developed to work on chunk of chromosomes.

Reply:
Ah then you don't have to do those checks to see if they are given :) you directly supply them to the module

Reply:
Ah true!
I will do so !